assets/src/components/mapMarkers.tsx

```diff
@@ -128,14 +144,34 @@ export const VehicleMarker = ({
   // > [...] if you want to put the marker on top of all others,
   // > [specify] a high value like 1000 [...]
   const zIndexOffset = isSelected ? 1000 : 0
+
+  // This offset is here because, due to a limitation of Leaflet
+  // popups, we weren't able to render the popup at the bottom-right
+  // corner of the marker, where it's supposed to go. This effectively
+  // renders it centered and above the marker, and then uses the
+  // offset to reposition it to the bottom-right corner.
+  const dropdownOffset = [140, 97]
+
   return (
     <>
       <Marker
         position={position}
         icon={vehicleIcon}
         eventHandlers={eventHandlers}
         zIndexOffset={zIndexOffset}
-      />
+        ref={markerRef}
+      >
+        {inTestGroup(TestGroups.DetoursPilot) && (
+          <Popup className="c-dropdown-popup-wrapper" offset={dropdownOffset}>
+            <DropdownMenu>
+              <DropdownItem>
+                Start a detour on route {vehicle.routeId}
+              </DropdownItem>
```

Comment:
question: have you thought about how we're going to get the callback that this option is selected? 
Also, _I think_, that once an option is selected, we'd need to close the popup, do you see that happening in another ticket?

Reply:
I envisioned having it close itself as part of the modal ticket, but .... actually maybe it would make more sense to put as part of this ticket, since it really is a behavior of the menu, not of the modal.

Reply:
@firestack These two commits are the approach I landed on - what do you think?
https://github.com/mbta/skate/pull/2373/commits/164175e24b59701c0ef0de19b8ee6853c3fa0714 <-- refactor
https://github.com/mbta/skate/pull/2373/commits/01b162362108627fcb5c2b9033da82533edd0be2 <-- ~1-liner to actually alert the on-click behavior

Reply:
I'm good with this!